src/AutoRest.CSharp/Common/Generation/Writers/FormattableStringHelpers.cs

```diff
@@ -145,12 +163,23 @@ private static bool IsContentTypeBinary(string contentType)
 
             var type = typeSubs[0];
             var subType = typeSubs[1];
+
+            if (subType == "json" && (type == "application" || type == "text"))
+            {
+                return BodyMediaType.Json;
+            }
+
             if (type == "audio" || type == "image" || type == "video" || subType == "octet-stream")
             {
-                return true;
+                return BodyMediaType.Binary;
+            }
+
+            if (type == "text")
+            {
+                return BodyMediaType.Text;
             }
 
-            return false;
+            throw new NotSupportedException($"Content type {contentType} is not supported.");
```

Comment:
What is the impact of throw exception here?  Will it cause the whole codegen to fail?  
If so, we should avoid that to happen.

Reply:
Yes, I would expect this to happen, because all the content types going to this line means they are not in the scopes that we have discussed, so we don't know what to generate. If exceptions come up then customers go to us and we will know that one is important.